doc/reference/configuration/configuration_reference.rst

```diff
@@ -2174,3 +2174,340 @@ The ``security`` section defines configuration parameters related to various sec
+    don't need to download all the data again.
+
+    Once all the nodes in the replica set are up and running,
+    automatic cleanup is started again even if ``wal.cleanup_delay`` has not expired.
+
+    ..  NOTE::
+
+        The option has no effect on nodes running as
+        :ref:`anonymous replicas <configuration_reference_replication_anon>`.
+
+    |
+    | Type: number
+    | Default: 14400
+    | Environment variable: TT_WAL_CLEANUP_DELAY
+
+..  _configuration_reference_wal_dir:
+
+..  confval:: wal.dir
+
+    A directory where write-ahead log (.xlog) files are stored. Can be relative
+    to ``process.work_dir``. Sometimes ``wal.dir`` and
+    :ref:`snapshot.dir <configuration_reference_snapshot_dir>` are specified with different values, so
+    that write-ahead log files and snapshot files can be stored on different
+    disks. If not specified, defaults to ``process.work_dir``.
+
+    |
+    | Type: string
+    | Default: 'var/lib/{{ instance_name }}'
+    | Environment variable: TT_WAL_DIR
+
+..  _configuration_reference_wal_dir_rescan_delay:
+
+..  confval:: wal.dir_rescan_delay
+
+    Number of seconds between periodic scans of the write-ahead-log
+    file directory, when checking for changes to write-ahead-log
+    files for the sake of :ref:`replication <replication>` or :ref:`hot standby <index-hot_standby>`.
+
+    |
+    | Type: number
+    | Default: 2
+    | Environment variable: TT_WAL_DIR_RESCAN_DELAY
+
+..  _configuration_reference_wal_max_size:
+
+..  confval:: wal.max_size
+
+    The maximum number of bytes in a single write-ahead log file.
+    When a request would cause an ``.xlog`` file to become larger than
+    ``wal.max_size``, Tarantool creates a new WAL file.
+
+    |
+    | Type: integer
+    | Default: 268435456
+    | Environment variable: TT_WAL_MAX_SIZE
+
+..  _configuration_reference_wal_mode:
+
+..  confval:: wal.mode
+
+    Specify fiber-WAL-disk synchronization mode as:
+
+    *   ``none``: write-ahead log is not maintained.
+        A node with ``wal.mode = none`` can't be replication master.
+
+    *   ``write``: :ref:`fibers <fiber-fibers>` wait for their data to be written to
+        the write-ahead log (no ``fsync(2)``).
+
+    *   ``fsync``: fibers wait for their data, :manpage:`fsync(2)`
+        follows each :manpage:`write(2)`.
+
+    |
+    | Type: string
+    | Default: 'write'
+    | Environment variable: TT_WAL_MODE
+
+..  _configuration_reference_wal_queue_max_size:
+
+..  confval:: wal.queue_max_size
+
+    The size of the queue in bytes used by a :ref:`replica <replication-roles>` to submit
+    new transactions to a :ref:`write-ahead log <internals-wal>` (WAL).
+    This option helps limit the rate at which a replica submits transactions to the WAL.
+    Limiting the queue size might be useful when a replica is trying to sync with a master and
+    reads new transactions faster than writing them to the WAL.
```

Comment:
How limiting the queue size helps in this scenario?

If WAL writes are going too slow on a replica, how more frequent flushing of the queue make thing better?